lib/psi.rb

```diff
@@ -61,79 +64,256 @@ def append(k)
 
 ##
 # Library of linear thermal bridges (e.g. corners, balconies). Each key:value
-# entry requires a unique identifier e.g. "poor (BC Hydro)" and a (partial or
+# entry requires a unique identifier e.g. "poor (BETBG)" and a (partial or
 # complete) set of PSI-values in W/K per linear meter.
 class PSI
   # @return [Hash] PSI set
+  # @return [Hash] shorthand listing of PSI types in a set
+  # @return [Hash] shorthand listing of PSI values in a set
   attr_reader :set
+  attr_reader :has
+  attr_reader :val
 
   ##
   # Construct a new PSI library (with defaults)
   def initialize
     @set = {}
-
-    # The following are defaults (* stated, ** inferred). Users may edit
-    # these sets, add new sets, or even read-in other sets on file.
-    # Units are in W/K per linear meter.
-    @set[ "poor (BC Hydro)" ] =
+    @has = {}
+    @val = {}
+
+    # The following are default PSI values (* published, ** calculated). Users
+    # may edit these sets, add new sets here, or read-in custom sets from a TBD
+    # JSON input file. PSI units are in W/K per linear meter. The spandrel sets
+    # are added as practical suggestions in early design stages.
+
+    # Convex vs concave PSI adjustments may be warranted if there is a mismatch
+    # between dimensioning conventions (interior vs exterior) used for the OSM
+    # vs published PSI data. For instance, the BETBG data reflects an interior
+    # dimensioning convention, while ISO 14683 reports PSI values for both
+    # conventions. The following may be used to adjust BETBG PSI values for
+    # convex corners when using outside dimensions for an OSM.
+    #
+    # PSIe = PSIi + U * 2(Li-Le), where:
+    #   PSIe = adjusted PSI                                          (W/K per m)
+    #   PSIi = initial published PSI                                 (W/K per m)
+    #      U = average clear field U-factor of adjacent walls           (W/m2.K)
+    #     Li = from interior corner to edge of "zone of influence"           (m)
+    #     Le = from exterior corner to edge of "zone of influence"           (m)
+    #
+    #  Li-Le = wall thickness e.g., -0.25m (negative here as Li < Le)
```

Comment:
A placeholder for now. This would need to find its way in future guide material.